pkg/webhook/deploy/deployment.go

```diff
@@ -46,6 +53,51 @@ func Webhook(ctx context.Context, cl runtimeclient.Client, s *runtime.Scheme, na
 	return nil
 }
 
+// Delete deletes the webhook app if it's deployed. Does nothing if it's not.
+// Returns true if the app was deleted.
+func Delete(ctx context.Context, cl runtimeclient.Client, s *runtime.Scheme, namespace string, oldObjectOnly bool) (bool, error) {
+	objs, err := getTemplateObjects(s, namespace, "0", []byte{00000001})
+	if err != nil {
+		return false, err
+	}
+
+	var deleted bool
+	for _, obj := range objs {
+		unst := &unstructured.Unstructured{}
+		unst.SetGroupVersionKind(obj.GetObjectKind().GroupVersionKind())
+		objName := obj.GetName()
+		_, doNotDeleteFound := obj.GetLabels()["toolchain.dev.openshift.com/no-deletion"]
+		if doNotDeleteFound {
+			// this object needs to stay
+			continue
+		}
+		// TODO --- temporary migration step to delete the objects by using the old name
+		if oldObjectOnly {
+			oldName, found := obj.GetLabels()["toolchain.dev.openshift.com/old-name"]
+			if !found {
+				// this object needs to stay
+				continue
+			}
+			objName = oldName
+		}
+		// TODO --- end temporary migration step
+		log.Info(fmt.Sprintf("Seraching for object to delete: %s  name:%s namespace:%s", obj.GetObjectKind().GroupVersionKind(), objName, obj.GetNamespace()))
+		if err := cl.Get(ctx, types.NamespacedName{Namespace: obj.GetNamespace(), Name: objName}, unst); err != nil {
+			if !errors.IsNotFound(err) { // Ignore not found
+				return false, errs.Wrap(err, "cannot get webhook object")
+			}
+		} else {
+			log.Info(fmt.Sprintf("Deleting %s  name:%s namespace:%s", obj.GetObjectKind().GroupVersionKind(), objName, obj.GetNamespace()))
+			if err := cl.Delete(ctx, unst); err != nil {
+				return false, errs.Wrap(err, "cannot delete webhook object")
+			}
+			deleted = true
+		}
```

Comment:
could be also:
```suggestion
			}
			continue
		}
		log.Info(fmt.Sprintf("Deleting %s  name:%s namespace:%s", obj.GetObjectKind().GroupVersionKind(), objName, obj.GetNamespace()))
		if err := cl.Delete(ctx, unst); err != nil {
			return false, errs.Wrap(err, "cannot delete webhook object")
		}
		deleted = true
```

Reply:
yup , I agree with removing `else` statements when possible.